sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/HL7v2MessageCoder.java

```diff
@@ -29,7 +29,15 @@
 import org.apache.beam.sdk.coders.StringUtf8Coder;
 
 public class HL7v2MessageCoder extends CustomCoder<HL7v2Message> {
-  public HL7v2MessageCoder() {}
+  HL7v2MessageCoder() {}
+
+  public static HL7v2MessageCoder of() {
+    return new HL7v2MessageCoder();
+  }
+
+  public static HL7v2MessageCoder of(Class<HL7v2Message> clazz) {
```

Comment:
Howcome you receive a class but ignore it?

Reply:
My understanding was this is necessary in order to use `@Default.Coder(HL7v2MessageCoder)` decorator on the `HL7v2Message` class.

However, I forgot to add that decorator so no wonder this looks confusing, good catch!
 
https://beam.apache.org/documentation/programming-guide/#default-coders-and-the-coderregistry

Reply:
Ah you're right  : ) - cool.